### PR TITLE
plugin: fix message-service

### LINKED ISSUE #7332: VSCodeAPI  'showInformationMessage' do not support newlines. (\n)
### Description

It should be possible for users to call `showInformationMessage` (VS Code API) with newlines included (`\n`). The following does not work for notification messages, but does work when passing [MessageOption](https://code.visualstudio.com/api/references/vscode-api#MessageOptions) `modal` as `true`.

### Reproduction Steps

1. create a simple vscode extension which calls `showInformationMessage` passing `modal: true` and include newlines in the message.

  Ex:

```ts
vscode.commands.registerCommand('extension.showModalMessage', () => {
  vscode.window.showInformationMessage('Hello World!\nHello World!\nHello World!', { modal: true });
})
```

**OS and Theia version:**

10.14.6
0.12.0

### LINKED ISSUE #7491: VScodeAPI: MessageItem "isCloseAffordance" does not hide "close" button in Theia.
### Description
When using VSCodeAPI to display an information message through `showInformationMessage`, the developer can use pass a list of `MessageItem` object containing a `title` (the button label) and an optional boolean `isCloseAffordance`. 

Closing the dialog without clicking on a button (like for example pressing ESC) will return the `MessageItem` which has `isCloseAffordance` set to true. Also when using `isCloseAffordance`, in VSCode it removes the "Cancel" button. However when using the same API on Theia, the modal does not hide its homologous "close" button.

### Reproduction Steps
- Use VSCode API method `showInformationMessage` with `MessageOption` `{modal:true}` and pass an array of `MessageItem`s with at least one of containing `isCloseAffordance:true`.
- Notice that in VSCode a modal shows up displaying only buttons added through the `MessageItem` array.  The "Cancel" buttons does not show.
- In Theia we still get the "close" button showing up.

**OS and Theia version:**
Mojave 10.14.6
Theia 0.12.0

### PR DESCRIPTION
**Pull request by vince-fugnitto**
#### What it does

Fixes #7332 
- The change fixes an issue where newlines were not respected when calling any `showMessage` (ex: `showInformationMessage`), when using the `modal` [MessageOption](https://code.visualstudio.com/api/references/vscode-api#MessageOptions).

Fixes #7491 
- The change updates the `showMessage` proxy to accept MessageItem type, and ultimately respect the `isCloseAffordance` option. Previously, the `close` button was always added to the dialog which did not match the behavior present in VS Code.

#### How to test

1. Start the application using the [message-tests](https://github.com/vince-fugnitto/message-tests/releases/download/1.0.0/message-tests-0.0.1.vsix) VS Code extension.
2. Execute the available commands, prefixed by `Message:`

---

Available Commands:

**`Show Message`**: 
displays a standard notification

**`Show Message (New Lines)`**: 
displays a standard notification message (new lines are unsupported in notifications

**`Show Modal Message`**: 
displays a modal dialog message

**`Show Modal Message (With Items)`**: 
displays a modal dialog message with an addition button

**`Show Modal Message (No Close)`**: 
displays a modal dialog message without the standard close button

**`Show Modal Message (No Close With Items)`**: 
displays a modal dialog message without the standard close button and with additional buttons

#### Review checklist

- [x] as an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

Signed-off-by: vince-fugnitto <vincent.fugnitto@ericsson.com>

### CODE REVIEW

**Comment by akosyakov** [comment]
@AlexTugarev Is it fine with you?

File: packages/plugin-ext/src/main/browser/dialogs/modal-notification.ts [hunk 1]
```diff
@@ -63,22 +64,24 @@ export class ModalNotification extends AbstractDialog<string | undefined> {
...
                 () => {
-                    this.actionTitle = action;
+                    this.actionTitle = action.title;
                     this.accept();
                 },
                 'click');
         });
-        this.appendCloseButton('close');
+        if (!actions.some(action => action.isCloseAffordance === true)) {
+            this.appendCloseButton('close');
```
Comment (AlexTugarev) [inline packages/plugin-ext/src/main/browser/dialogs/modal-notification.ts:83]: nit: in most cases first upper case is used. this isn't changed here, I just saw this while testing

Comment (vince-fugnitto) [reply packages/plugin-ext/src/main/browser/dialogs/modal-notification.ts:83]: > nit: in most cases first upper case is used. this isn't changed here, I just saw this while testing

@AlexTugarev do you mind clarifying? (what sort of change is required)

Comment (vince-fugnitto) [reply packages/plugin-ext/src/main/browser/dialogs/modal-notification.ts:83]: Ping :)

Comment (AlexTugarev) [reply packages/plugin-ext/src/main/browser/dialogs/modal-notification.ts:83]: missed that ;-)
it was not important. `Close` would be better aligned with other button titles. nvm.

Comment (vince-fugnitto) [reply packages/plugin-ext/src/main/browser/dialogs/modal-notification.ts:83]: No problem, thank you for the review! 😄

**Comment by vince-fugnitto** [comment]
@benoitf any breaking changes for che?

**Comment by akosyakov** [comment]
@vince-fugnitto are you going to merge it? if we break something we can revert/fix it with a follow-up PR

**Comment by vince-fugnitto** [comment]
> @vince-fugnitto are you going to merge it? if we break something we can revert/fix it with a follow-up PR

Thank you, I'll merge it, it's been in review a while now for others to see.

**Pull request merged**